tests/baselines/reference/moduleResolution/baseUrl-without-path-mappings-or-rootDirs.js

```diff
@@ -13,10 +13,7 @@ Resolution:: {
     "resolvedFileName": "/root/folder2/file2.ts",
     "extension": ".ts",
     "isExternalLibraryImport": false
-  },
-  "failedLookupLocations": [],
-  "affectingLocations": [],
-  "resolutionDiagnostics": []
```

Comment:
I always assumed these were non-optional for V8 monomorphism reasons (though I would expect `undefined` instead of an empty array might be the best for performance?). @DanielRosenwasser @rbuckton is there a guiding principle for how to approach these things? Does it even make sense to try to make optimizations without looking at data from deoptigate?

Reply:
Made these optional because we want to handle optional undefined. We can always set these to empty array like before but thought undefined instead of empty array would better (since it is mutable empty array so always new array)

Reply:
Yeah, I’m sure the new arrays is the worst of all the options. There might be some benefit to initializing to `undefined` but I’m not sure.